client/src/modules/reports/generate/budget_report/budget_report.config.js

```diff
@@ -0,0 +1,78 @@
+angular.module('bhima.controllers')
+  .controller('budget_reportController', BudgetReportController);
+
+BudgetReportController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state', 'SessionService',
+];
+
+function BudgetReportController($sce, Notify, SavedReports, AppCache, reportData, $state, Session) {
+  const vm = this;
+  const cache = new AppCache('configure_budget_report');
+  const reportUrl = 'reports/finance/budget_report';
+
+  vm.reportDetails = {
+    currency_id : Session.enterprise.currency_id,
+  };
+
+  vm.previewGenerated = false;
+  checkCachedConfiguration();
+
+  vm.onSelectFiscalYear = (fiscalYear) => {
+    vm.reportDetails.fiscal_id = fiscalYear.id;
+  };
+
+  vm.onSelectCurrency = (currency) => {
+    vm.reportDetails.currency_id = currency.id;
+  };
+
+  vm.onSelectCronReport = report => {
+    vm.reportDetails = angular.copy(report);
+  };
+
+  vm.preview = function preview(form) {
+    if (form.$invalid) { return null; }
+
+    // update cached configuration
+    cache.reportDetails = angular.copy(vm.reportDetails);
+
+    return SavedReports.requestPreview(reportUrl, reportData.id, angular.copy(vm.reportDetails))
```

Comment:
This error handling doesn't seem to work if there is no budget for that year.  I tried it for the test dataset with FY 2019, and it failed with this error, but there was not signal given on the client.

![image](https://github.com/user-attachments/assets/641e08ab-3d80-4c66-b9db-9f2e5218171f)

It should probably let the user know that no budget was set for that fiscal year.

Reply:
> This error handling doesn't seem to work if there is no budget for that year. I tried it for the test dataset with FY 2019, and it failed with this error, but there was not signal given on the client.
> 
> ![image](https://private-user-images.githubusercontent.com/896472/365454983-641e08ab-3d80-4c66-b9db-9f2e5218171f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1NTA4NzQsIm5iZiI6MTcyNjU1MDU3NCwicGF0aCI6Ii84OTY0NzIvMzY1NDU0OTgzLTY0MWUwOGFiLTNkODAtNGM2Ni1iOWRiLTlmMmU1MjE4MTcxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxN1QwNTIyNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDRhMjYzN2ViZTcxMjA5MjI0NmI4MTI1YTY3ZmU2ZTg0NzYwZjJiZWJmMmNiNTA5YjQyZTBmYWU4ODU2ODFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.WxO3dLjjyx7d8jhffvyAkP3qDX8KbpCuiUsrVdn4PgM)
> 
> It should probably let the user know that no budget was set for that fiscal year.
I can't reproduce the error, I used the test database by first running ````npm run build:db````
![image](https://github.com/user-attachments/assets/16a41ff8-7603-44a4-b5f5-1e95d3aec9a8)

Reply:
![image](https://github.com/user-attachments/assets/30317773-756e-49fd-9e67-20429eaa7dc0)

Reply:
This seems to be resolved.   I tested several FY's and they all worked.  If there was no budget, 0 was used for all items.